lib/builtin.pm

```diff
@@ -122,13 +122,14 @@ currently but will be removed in a later version.
 Weakens a reference. A weakened reference does not contribute to the reference
 count of its referent. If only weakened references to a referent remain, it
 will be disposed of, and all remaining weak references to it will have their
-value set to C<undef>.
+value set to C<undef>.  Dies when provided with a defined non-reference.
```

Comment:
Hrm, it's all a bit weird here. This distinction goes right the way down to the actual `sv_rvweaken` function, which complains about defined but non-referential values, but lets undef itself have a special pass and just ignores it. That just is what the function has always done, but it still feels kinda weird.

Given that, I wonder if we should specifically also point out that undef is permitted but ignored.

Also I tend to avoid the verb "die" in actual documentation - that's the perl operator name for it, but I usually describe it as "throws an exception".